### PR TITLE
Convert actions to use OIDC deployment pattern

### PR DESCRIPTION
**Pull request by ividito**
Completes the last item of #292 - this PR fully switches us to using the pattern defined in #312

### CODE REVIEW

File: .github/workflows/deploy_oidc.yml [hunk 1]
```diff
@@ -15,11 +17,29 @@ on:
...
+    permissions:
+      id-token: write
+      contents: read
     environment: ${{ inputs.deploy_env }}
     steps:
       - name: Check out repo
         uses: actions/checkout@v3
 
+      - name: Configure awscli (local runner)
+        if: ${{ env.ACT }} # OIDC doesn't work when using ACT, as AWS user accounts don't have access to the OIDC role by default
```
Comment (slesaad) [inline .github/workflows/deploy_oidc.yml:29]: what's ACT?

File: .github/workflows/cicd.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 name: CI/CD
 
 on:
-  push:
+  workflow_dispatch:
```
Comment (slesaad) [inline .github/workflows/cicd.yml:4]: why'd we change this?

File: .github/workflows/deploy_oidc.yml (hunk 1 above)
Comment (ividito) [reply .github/workflows/deploy_oidc.yml:29]: This is automatically set when running the action locally using https://github.com/nektos/act

Comment (slesaad) [reply .github/workflows/deploy_oidc.yml:29]: ah makes sense. thanks

**Review by slesaad — APPROVED** [review]
LGTM

**Pull request merged**